### ISSUE TITLE
autodoc: :members: list causes :special-members: not to be shown

### ISSUE THREAD

**Issue by andy-maier** [sphinx-doc/sphinx#2401]
In a `autoclass` statement, when its `:members:` option specifies a list of members to show, the special members are no longer shown.

This fragment in index.rst will show all public non-special members and the two listed special members in the output:

```
.. autoclass:: mypackage.MyClass
   :members:
   :special-members: __str__, __repr__
```

While this one produces the failure by showing the specified public members but no more special members in the output:

```
.. autoclass:: mypackage.MyClass
   :members: method1, method2
   :special-members: __str__, __repr__
```

I am using Sphinx 1.4 on CentOS 6.7.
Let me know in case you need more data.

**Comment by tk0miya** [comment]
It seems the argument for  `:special-members:` is ignored when the argument for `:members:` is given.
I think this is a bug of autodoc module.

As a workaround, you can specify special members to `:members:`:
```
.. autoclass:: mypackage.MyClass
   :members: method1, method2, __str__, __repr__
```

**Comment by tk0miya** [comment]
Fixed by #5376.
Thank you for reporting.
